cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerLoggingConfigurationDiff.java

```diff
@@ -179,4 +104,47 @@ private static boolean isValidLoggerLevel(String level) {
         return VALID_LOGGER_LEVELS.contains(level);
     }
 
+    static class LoggingLevelResolver {
+
+        Map<String, String> config;
+
+        LoggingLevelResolver(Map<String, String> loggingConfig) {
+            this.config = loggingConfig;
+        }
+
+        LoggingLevel resolveLevel(String name) {
+            String level = config.get(name);
+            if (level != null) {
+                return LoggingLevel.valueOf(level.split(",")[0]);
+            }
+
+            int e = name.length();
+            int b = e;
+            while (b > -1) {
+                b = name.lastIndexOf('.', e);
+                if (b == -1) {
+                    level = config.get("root");
+                } else {
+                    level = config.get(name.substring(0, b));
+                }
+                if (level != null) {
+                    return LoggingLevel.valueOf(level.split(",")[0]);
+                }
+                e = b - 1;
+            }
+            // still here? Not even root logger defined?
+            return LoggingLevel.INFO;
+        }
+    }
+
+    enum LoggingLevel {
+        ALL,
+        FATAL,
+        ERROR,
+        WARN,
+        INFO,
+        DEBUG,
+        TRACE,
+        OFF
```

Comment:
See point 4 in rejected alternatives:
https://cwiki.apache.org/confluence/display/KAFKA/KIP-412%3A+Extend+Admin+API+to+support+dynamic+application+log+levels#KIP412:ExtendAdminAPItosupportdynamicapplicationloglevels-RejectedAlternatives

Reply:
One can type anything in Kafka CR logging / loggers, including otherwise valid levels like FINE and FINEST which are perfectly valid for java util logging, but not for Log4j. The question is how do we want to proceed when an unsupported logging level is specified. Currently FINE and FINEST probably just break things so we have to add some exception catching.
We can remove OFF from enum, and handle it the same way as FINE, FINEST.

Reply:
Actually the KIP-412 also does not seem to understand ALL level, so we should treat it as TRACE.